pkg/api/v1/router_bom.go

```diff
@@ -0,0 +1,80 @@
+package serverservice
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+
+	"go.hollow.sh/serverservice/internal/models"
+)
+
+// TODO: implement abortCleanup and call it during the early abort.
+// abortCleanup will delete already inserted data if error happens in the middle of the upload.
+// Otherwise users may retry uploading the file again while some bom has already been uploaded. (Or should we override any existing row?)
+// func (r *Router) abortCleanup() {}
+
+func (r *Router) bomsUpload(c *gin.Context) {
+	var boms []Bom
+	if err := c.ShouldBindJSON(&boms); err != nil {
+		badRequestResponse(c, "invalid payload: []Bom{}", err)
+		return
+	}
+
+	for _, bom := range boms {
```

Comment:
This is better done in a transaction - so its all data or nothing, what do you think?

Reply:
I am with you. Updated the code to use transaction. Thanks for the suggestion Joel.